### PR TITLE
pass along hooks, add test to server impl

### PR DESCRIPTION
**Pull request by sveisvei**
Spawned from https://github.com/Unleash/unleash/issues/222

### CODE REVIEW

File: lib/server-impl.js [hunk 1]
```diff
@@ -21,10 +21,14 @@ function createApp (options) {
         enableRequestLogger: options.enableRequestLogger,
         port: options.port,
         publicFolder: options.publicFolder,
+        preRouterHook: options.preRouterHook,
+        preHook: options.preHook,
         stores,
         eventBus,
     };
 
+    console.log(config);
```
Comment (SimenB) [inline lib/server-impl.js]: no

Comment (sveisvei) [inline lib/server-impl.js]: That never happened 🔡

File: lib/server-impl.js [hunk 2]
```diff
@@ -21,6 +21,8 @@ function createApp (options) {
         enableRequestLogger: options.enableRequestLogger,
         port: options.port,
         publicFolder: options.publicFolder,
+        preRouterHook: options.preRouterHook,
```
Comment (SimenB) [inline lib/server-impl.js]: should this whole `config` pass everything through?

Comment (sveisvei) [inline lib/server-impl.js]: Cannot se a reason for why not - @ivarconr ? Changed to use object.assign in another commit.

File: lib/server-impl.js [hunk 3]
```diff
@@ -15,15 +15,10 @@ function createApp (options) {
     const stores = createStores(options);
```
Comment (SimenB) [inline lib/server-impl.js]: pass it into here?

Comment (sveisvei) [inline lib/server-impl.js]: Was what?

Comment (SimenB) [inline lib/server-impl.js]: `config` created below. It now passes in `options` before creating `config` with it.

But nevermind, it'll be the same result as it is now.

**Pull request merged**